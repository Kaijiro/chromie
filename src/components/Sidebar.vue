<template>
  <ul>
    <li v-for="encounter in encounters">{{ encounter.name }}</li>
  </ul>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import {mapState} from "vuex";

export default defineComponent({
  computed: mapState({
    encounters: 'encounters'
  })
});
</script>
