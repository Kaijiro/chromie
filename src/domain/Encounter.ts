export default class Encounter {
    private fighters: Array<string> = [];

    countFighters(): number {
        return this.fighters.length;
    }

    addFighter(myFighter: string) {
        this.fighters.push(myFighter);
    }
}
