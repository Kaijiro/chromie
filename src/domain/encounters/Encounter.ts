export default class Encounter {
    private readonly name: string;
    private fighters: Array<string> = [];

    private constructor(name: string) {
        this.name = name;
    }

    public static with(name: string): Encounter{
        return new Encounter(name);
    }

    countFighters(): number {
        return this.fighters.length;
    }

    addFighter(fighter: string) {
        this.fighters.push(fighter);
    }

    getName() {
        return this.name;
    }
}
