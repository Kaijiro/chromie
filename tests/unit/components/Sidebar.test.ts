import {shallowMount} from "@vue/test-utils";
import Sidebar from "../../../src/components/Sidebar.vue";
import Encounter from "../../../src/domain/encounters/Encounter";

describe("Sidebar component", () => {
    it(`should display a list of all encounters`, () => {
        const mockStore = {
            state: {
                encounters: [new Encounter("Boss n°1"), new Encounter("Boss n°2"), new Encounter("Boss n°3")]
            }
        };
        const component = shallowMount(Sidebar, {
            global: {
                mocks: {
                    $store: mockStore
                }
            }
        });

        // It's currently a bit dirty to access these private attributes but this is the only way right now
        expect(component["__app"]._container).toMatchSnapshot();
    });
});
