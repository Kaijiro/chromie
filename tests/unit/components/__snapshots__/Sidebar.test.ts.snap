// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should display a list of all encounters 1`] = `
<div
  data-v-app=""
>
  <ul>
    
    <li>
      Boss n°1
    </li>
    <li>
      Boss n°2
    </li>
    <li>
      Boss n°3
    </li>
    
  </ul>
</div>
`;
